### PR TITLE
Add reminder function to filter by days to interview date

### LINKED ISSUE #99: Add command to display interviews in the upcoming n days

### PR DESCRIPTION
**Pull request by Ditzchann**
Fixes #99

### CODE REVIEW

File: src/main/java/hirehive/address/logic/commands/FilterUtil.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     * @param model The model to update the list in
+     * @param predicate The given predicate to filter the list with
+     * @return A CommandResult object
+     */
+    public static CommandResult executeFilter(Model model, Predicate<Person> predicate) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(predicate);
+        String message;
+        if (!model.getFilteredPersonList().isEmpty()) {
+            message = String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size());
```
Comment (m0destly) [inline src/main/java/hirehive/address/logic/commands/FilterUtil.java:28]: Since we have the fixed counter already, this might be a bit repetitive. I would suggest returning a message that says "showing all [t/TAG] persons" since the "command" disappears after valid input is accepted.

File: src/main/java/hirehive/address/logic/commands/FilterCommand.java [hunk 2]
```diff
@@ -29,15 +29,7 @@ public FilterCommand(PersonContainsTagPredicate predicate) {
...
-        requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        String message;
-        if (!model.getFilteredPersonList().isEmpty()) {
-            message = String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size());
-        } else {
-            message = Messages.MESSAGE_NO_SUCH_PERSON;
-        }
-        return new CommandResult(message);
+        return FilterUtil.executeFilter(model, predicate);
```
Comment (m0destly) [inline src/main/java/hirehive/address/logic/commands/FilterCommand.java:32]: Did you abstract the original Filter execute method into FilterUtil that calls a single method executeFilter to reuse the method for Reminder?

File: src/main/java/hirehive/address/commons/util/StringUtil.java [hunk 3]
```diff
@@ -60,4 +60,22 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+    public static boolean isPositiveInteger(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            return value >= 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+    }
```
Comment (m0destly) [inline src/main/java/hirehive/address/commons/util/StringUtil.java:80]: Good use of the Util class to add helper methods

File: src/test/java/hirehive/address/commons/util/StringUtilTest.java [hunk 4]
```diff
@@ -47,6 +47,35 @@ public void isNonZeroUnsignedInteger() {
...
+        assertFalse(StringUtil.isPositiveInteger(" 10 ")); // Leading/trailing spaces
+        assertFalse(StringUtil.isPositiveInteger("1 0")); // Spaces in the middle
+
+        // EP: number larger than Integer.MAX_VALUE
+        assertFalse(StringUtil.isPositiveInteger(Long.toString(Integer.MAX_VALUE + 1)));
+
+        // EP: valid numbers, should return true
+        assertTrue(StringUtil.isPositiveInteger("0"));
+        assertTrue(StringUtil.isPositiveInteger("10"));
+    }
```
Comment (m0destly) [inline src/test/java/hirehive/address/commons/util/StringUtilTest.java:77]: Great that you updated the tests, and also made use of boundary values to test your code rigorously.

**Review by m0destly — APPROVED** [review]
Can add a few more tests just to cover the last few lines of code coverage, otherwise LGTM

File: src/main/java/hirehive/address/logic/commands/FilterCommand.java (hunk 2 above)
Comment (Ditzchann) [reply src/main/java/hirehive/address/logic/commands/FilterCommand.java:32]: Yep! Any future similar "filter" functions can make use of this function as well!
